File: CS/SchedulerModel.cs
using System;

namespace SchedulerBindToObservableCollectionWpf {
#region #models    
    public class ModelAppointment {
        public object Id { get; set; }
        public DateTime StartTime { get; set; }
        public DateTime EndTime { get; set; }
        public string Subject { get; set; }
        public int Status { get; set; }
        public string Description { get; set; }
        public int Label { get; set; }
        public string Location { get; set; }
        public bool AllDay { get; set; }
        public int EventType { get; set; }
        public string RecurrenceInfo { get; set; }
        public string ReminderInfo { get; set; }
        public object ResourceId { get; set; }
        public decimal Price { get; set; }
        public ModelAppointment() {
        }
    }

    public class ModelResource {
        public int Id { get; set; }
        public string Name { get; set; }
        public int Color { get; set; }

        public ModelResource() {
        }
    }
#endregion #models
}

File: VB/SchedulerModel.vb
Imports System

Namespace SchedulerBindToObservableCollectionWpf
#Region "#models    "
	Public Class ModelAppointment
		Public Property Id() As Object
		Public Property StartTime() As DateTime
		Public Property EndTime() As DateTime
		Public Property Subject() As String
		Public Property Status() As Integer
		Public Property Description() As String
		Public Property Label() As Integer
		Public Property Location() As String
		Public Property AllDay() As Boolean
		Public Property EventType() As Integer
		Public Property RecurrenceInfo() As String
		Public Property ReminderInfo() As String
		Public Property ResourceId() As Object
		Public Property Price() As Decimal
		Public Sub New()
		End Sub
	End Class

	Public Class ModelResource
		Public Property Id() As Integer
		Public Property Name() As String
		Public Property Color() As Integer

		Public Sub New()
		End Sub
	End Class
#End Region ' #models
End Namespace
